internal/syscallfs/dirfs.go

```diff
@@ -24,7 +23,7 @@ type dirFS string
 
 // Open implements the same method as documented on fs.FS
 func (dir dirFS) Open(name string) (fs.File, error) {
-	panic(fmt.Errorf("unexpected to call fs.FS.Open(%s)", name))
+	return dir.OpenFile(name, 0, 0)
```

Comment:
I changed this when the `syscallfs.FS` type was still declaring that it implemented `fs.FS` so that the implementations could pass the `fstest.TestFS` test suite, but I can revert it if we don't want to validate this as part of the pull request.

Reply:
I think for this we can add a test-only fs wrapper instead, since the purpose is only for tests using `fstest`. In that, we'd have to follow the fs.FS rules like cleaning the path and making sure it is relative as iirc they will test them.

We can add to that test wrapper each interface we use internally besides `fs.FS`. For example, right now, we don't use anything in adapter.go except `fs.FS`, but if we did use something else, we can implement that.